manager/controllers/app/m4dapplication_controller.go

```diff
@@ -249,6 +249,22 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+	// Store user credentials in vault
+	if applicationContext.Spec.AppInfo.UserSecretRef != "" {
+		var data map[string]interface{}
+		if data, err = SecretToCredentialMap(r.Client,
+			types.NamespacedName{Name: applicationContext.Spec.AppInfo.UserSecretRef, Namespace: applicationContext.Namespace}); err != nil {
+			r.Log.V(0).Info("Error reading user secret: " + err.Error())
+			setCondition(applicationContext, "", err.Error(), true)
+			return ctrl.Result{}, err
+		}
+		path := utils.GenerateUserCredentialsSecretName(applicationContext.Namespace, applicationContext.Name, utils.GetCatalogProviderName())
```

Comment:
why is catalog provider name needed? Why do we care what the catalog system is?

Reply:
I've tried to imitate what we have already. There is a naming convention for the vault path (keeping user credentials) which consists of prefix/appNamespace/appName/Catalog, e.g. m4d/user-creds/default/notebook/Egeria. This was used by the GUI.

Reply:
CHANGE: catalog system is no longer relevant. Each system name is a key in the secret.